cmd/kanikoExecute.go

```diff
@@ -92,6 +93,18 @@ func runKanikoExecute(config *kanikoExecuteOptions, telemetryData *telemetry.Cus
 		config.BuildOptions = append(config.BuildOptions, dest...)
 	}
 
+	var dataParametersJSON map[string]interface{}
+	var errUnmarshal = json.Unmarshal([]byte(GeneralConfig.ParametersJSON), &dataParametersJSON)
+	if errUnmarshal != nil {
+		log.Entry().Infof("Reading ParametersJSON is failed")
+	}
+	if value, ok := dataParametersJSON["dockerImage"]; ok {
+		commonPipelineEnvironment.custom.dockerImage = value.(string)
+	} else {
+		dataStepMetadata := GetAllStepMetadata()
```

Comment:
`GetAllStepMetadata()` only resolves the defaults of the step but not the actual configuration.
Result will be the wrong image in many cases.

Resolution would need to get through the same logic as the call `piper getConfig --contextConfig`.
This may also have additional dependencies on custom defaults.
Thus, it might be a bit tricky to get the proper config from within the step since this is typically nothing the step would know so far.

One option I could think of would be to enhance the step config resolution in a way that the dockerImage is always available via the respective config struct.